vision/build.gradle

```diff
@@ -9,92 +9,146 @@ plugins {
-	implementation "edu.wpi.first.apriltag:apriltag-java:$wpilibVersion"
+	annotationProcessor wpi.java.deps.wpilibAnnotations()
+	implementation wpi.java.deps.wpilib()
+	implementation wpi.java.vendor.java()
 
-	//def akitJson = new groovy.json.JsonSlurper().parseText(new File(projectDir.getAbsolutePath() + "/vendordeps/AdvantageKit.json").text)
-	//annotationProcessor "org.littletonrobotics.akit.junction:junction-autolog:$akitJson.version"
-	def akitVersion = "4.0.0-alpha-1"
-	annotationProcessor "org.littletonrobotics.akit.junction:junction-autolog:$akitVersion"
+	roborioDebug wpi.java.deps.wpilibJniDebug(wpi.platforms.roborio)
+	roborioDebug wpi.java.vendor.jniDebug(wpi.platforms.roborio)
 
-	implementation "org.littletonrobotics.akit.junction:junction-core:$akitVersion"
+	roborioRelease wpi.java.deps.wpilibJniRelease(wpi.platforms.roborio)
+	roborioRelease wpi.java.vendor.jniRelease(wpi.platforms.roborio)
 
-	def photonVersion = "v2025.0.0-beta-5"
-	implementation "org.photonvision:photonlib-java:$photonVersion"
-	implementation "org.photonvision:photontargeting-java:$photonVersion"
+	nativeDebug wpi.java.deps.wpilibJniDebug(wpi.platforms.desktop)
+	nativeDebug wpi.java.vendor.jniDebug(wpi.platforms.desktop)
+	simulationDebug wpi.sim.enableDebug()
 
+	nativeRelease wpi.java.deps.wpilibJniRelease(wpi.platforms.desktop)
+	nativeRelease wpi.java.vendor.jniRelease(wpi.platforms.desktop)
+	simulationRelease wpi.sim.enableRelease()
+
+	testImplementation 'org.junit.jupiter:junit-jupiter:5.10.1'
 	testRuntimeOnly 'org.junit.platform:junit-platform-launcher'
 
-	implementation 'com.googlecode.json-simple:json-simple:1.1.1'
+	def akitJson = new groovy.json.JsonSlurper().parseText(new File(projectDir.getAbsolutePath() + "/vendordeps/AdvantageKit.json").text)
+	annotationProcessor "org.littletonrobotics.akit:akit-autolog:$akitJson.version"
 }
 
 test {
 	useJUnitPlatform()
 	systemProperty 'junit.jupiter.extensions.autodetection.enabled', 'true'
 }
 
-// Configure string concat to always inline compile
-tasks.withType(JavaCompile) {
-	options.compilerArgs.add '-XDstringConcat=inline'
+// Simulation configuration (e.g. environment variables).
+//
+// The sim GUI is *disabled* by default to support running
+// AdvantageKit log replay from the command line. Set the
+// value to "true" to enable the sim GUI by default (this
+// is the standard WPILib behavior).
+wpi.sim.addGui().defaultEnabled = false
+wpi.sim.addDriverstation()
+
+// Setting up my Jar File. In this case, adding all libraries into the main jar ('fat jar')
+// in order to make them all available at runtime. Also adding the manifest so WPILib
+// knows where to look for our Robot Class.
+jar {
+	from { configurations.runtimeClasspath.collect { it.isDirectory() ? it : zipTree(it) } }
+	from sourceSets.main.allSource
+	manifest edu.wpi.first.gradlerio.GradleRIOPlugin.javaManifest(ROBOT_MAIN_CLASS)
+	duplicatesStrategy = DuplicatesStrategy.INCLUDE
```

Comment:
we shouldn't need to do this (but we'll need to verify via integrating in 2025 robot code). The maven publisher should handle making the jar file.

If it's a problem to not have this, we will need to find a creative way to avoid GradleRIO

Reply:
still builds without this, testing in 2025 soon